contracts/lib/FulfillmentApplier.sol

```diff
@@ -528,244 +484,167 @@ contract FulfillmentApplier is FulfillmentApplicationErrors {
+                // If the consideration item index is out of range or the numerator
+                // is zero, skip this item.
+                if or(
+                    iszero(lt(itemIndex, mload(considerationArrPtr))),
+                    iszero(mload(add(orderPtr, AdvancedOrder_numerator_offset)))
+                ) {
+                    continue
                 }
 
-                // Retrieve consideration item pointer using index.
-                considerationItemPtr := mload(
+                // Retrieve consideration item pointer using the item index.
+                let considerationItemPtr := mload(
                     add(
                         // Get pointer to beginning of receivedItem.
                         add(considerationArrPtr, OneWord),
-                        // Use offset to pointer for desired order.
+                        // Calculate offset to pointer for desired order.
                         mul(itemIndex, OneWord)
                     )
                 )
 
-                // Retrieve amount pointer using consideration item pointer.
-                let amountPtr := add(
-                      considerationItemPtr,
-                      Common_amount_offset
-                )
+                // Declare a separate scope for the amount update
+                {
+                    // Retrieve amount pointer using consideration item pointer.
+                    let amountPtr := add(
+                        considerationItemPtr,
+                        Common_amount_offset
+                    )
 
-                // Add offer amount to execution amount.
-                let newAmount := add(amount, mload(amountPtr))
+                    // Add offer amount to execution amount.
+                    let newAmount := add(amount, mload(amountPtr))
 
-                // Update error buffer: 1 = zero amount, 2 = overflow, 3 = both.
-                errorBuffer := or(
-                  errorBuffer,
-                  or(
-                    shl(1, lt(newAmount, amount)),
-                    iszero(mload(amountPtr))
-                  )
-                )
+                    // Update error buffer: 1 = zero amount, 2 = overflow, 3 = both
+                    errorBuffer := or(
+                        errorBuffer,
+                        or(
+                            shl(1, lt(newAmount, amount)),
+                            iszero(mload(amountPtr))
+                        )
+                    )
 
-                // Update the amount to the new, summed amount.
-                amount := newAmount
+                    // Update the amount to the new, summed amount.
+                    amount := newAmount
 
-                // Zero out amount on original item to indicate it is credited.
-                mstore(amountPtr, 0)
+                    // Zero out amount on original item to indicate it is credited.
+                    mstore(amountPtr, 0)
+                }
```

Comment:
For offer item aggeregation subroutine this part comes after the `switch` block.

Reply:
It would be great to keep the flow consistent for both functions.

Reply:
It is important to keep this block in this function before the `switch` block, due to the hashing comparison that requires the `amount` slot to be zeroed out.